staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+)
+
+// SubresourceScaler manages the conversion of managed fields between a main
+// resource and subresources
+type SubresourceScaler struct {
+	parentEntries           []metav1.ManagedFieldsEntry
+	parentGroupVersion      schema.GroupVersion
```

Comment:
I'm pretty sure this is irrelevant, and depends on the version of each individual managedFields that you're looking at. Especially since the scaler here doesn't look at the object, but merely at the managedFields. You can probably remove this field.

Reply:
I have added it here because we need to convert back the updated managed fields into the correct API Version, I am not sure how to achieve that otherwise 🤔